packages/graphback-core/src/runtime/GraphbackDataProvider.ts

```diff
@@ -22,7 +22,7 @@ export interface GraphbackDataProvider<Type = any, GraphbackContext = any> {
    * @param data input data
    * @param context context object passed from graphql or rest layer
    */
-  create(data: Type, context?: GraphbackContext): Promise<Type>;
+  create(data: Type, context: GraphbackContext): Promise<Type>;
```

Comment:
I have a concern with passing the entire context object down to the data provider with every request. All that is used is the selectedFields.

Context can have:

- A CRUD service for every model
- A data provider for every model
- 0...N data loaders
- Request/response data
- Selected fields
- Other unknown user context data

This seems too much to pass down to the provider for the purpose of using only a tiny fraction of this.

Reply:
Objects are passed by reference and context is a general-purpose function that is widely used across requests. We have been silently using it for many purposes so it will make sense to standardize it and use it even more :D

Reply:
Good point.